features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentEdit.jsx

```diff
@@ -0,0 +1,230 @@
+    }
+
+    /**
+     * Handles the comment text when input changes
+     * @param {Object} {target} target element
+     * @memberof CommentEdit
+     */
+    inputChange({ target }) {
+        this.setState({ commentText: target.value, currentLength: target.value.length });
+    }
+
+    /**
+     * Hides the component to edit a comment when cancelled
+     * @memberof CommentEdit
+     */
+    handleClickCancel() {
+        const { toggleShowEdit, commentsUpdate, allComments } = this.props;
+        toggleShowEdit();
+        commentsUpdate(allComments);
+    }
+
+    /**
+     * Handles category when the category is changed
+     * @param {any} event Drop down select event
+     * @memberof CommentEdit
+     */
+    handleCategoryChange(event) {
+        this.setState({ category: event.target.value });
+    }
+
+    /**
+     * Filters the comment to update
+     * @memberof CommentAdd
+     */
+    filterCommentToUpdate(commentToFilter) {
+        const { comment } = this.props;
+        return commentToFilter.commentId === comment.commentId;
+    }
+
+    /**
+     * Filters the comment to update
+     * @memberof CommentAdd
+     */
+    filterCommentToUpdateReply(commentToFilter) {
+        const { comment } = this.props;
+        return commentToFilter.commentId === comment.parentCommentId;
+    }
+
+    /**
+     * Handles updating a comment
+     * @memberof CommentEdit
+     */
+    handleClickUpdateComment() {
+        const {
+            api, comment, allComments, toggleShowEdit, commentsUpdate,
+        } = this.props;
+        const { category, commentText } = this.state;
+        const Api = new API();
+        const apiId = api.id;
+        const commentToEdit = comment;
+        commentToEdit.commentText = commentText.trim();
+        commentToEdit.category = category;
+
+        // to check whether a string does not contain only whitehis spaces
+        if (comment.commentText.replace(/\s/g, '').length) {
+            Api.updateComment(apiId, commentToEdit.commentId, commentToEdit)
+                .then((result) => {
+                    const updatedComment = result.body;
+                    if (commentToEdit.parentCommentId === undefined) {
+                        const index = allComments.findIndex(this.filterCommentToUpdate);
+                        allComments[index].category = updatedComment.category;
+                        allComments[index].commentText = updatedComment.commentText;
+                    } else {
+                        const index = allComments.findIndex(this.filterCommentToUpdateReply);
+                        const replyIndex = allComments[index].replies.findIndex(this.filterCommentToUpdate);
+                        allComments[index].replies[replyIndex] = updatedComment;
+                    }
+                    toggleShowEdit();
+                    commentsUpdate(allComments);
+                })
+                .catch((error) => {
+                    console.error(error);
```

Comment:
can we remove this console statement ?

Reply:
We are using console.error and console.log to log errors through out the store and publisher UIs. Whats the alternative you are proposing?

Reply:
So you are suggesting not to remove the console statement and put a different error message?

In that case an user who needs to get the actual error message has to debug and read the real one from browser console. If there is a way to satisfy both of these use-cases it's nice. 

From the security point of view the error returning to here is a broken promise exception with a reason why the promise is broken. The reason will be the exception that will be return as the error response from the API call. So as long as the Rest API is developer is not return any sensitive information( which is not happening ) , it's fine IMO.